internal/postgresql/postgresql.go

```diff
@@ -299,10 +299,98 @@ func (p *Manager) StartTmpMerged() error {
 	return p.start("-c", fmt.Sprintf("config_file=%s", tmpPostgresConfPath))
 }
 
+func (p *Manager) moveWal() (err error) {
+	var curPath string
+	var desiredPath string
+	var tmpPath string
+	symlinkPath := filepath.Join(p.dataDir, "pg_wal")
+	if curPath, err = filepath.EvalSymlinks(symlinkPath); err != nil {
+		log.Errorf("could not evaluate symlink %s: %e", symlinkPath, err)
+		return err
+	}
+	if p.walDir == "" {
+		desiredPath = symlinkPath
+		tmpPath = filepath.Join(p.dataDir, "pg_wal_new")
+	} else {
+		desiredPath = p.walDir
+		tmpPath = p.walDir
+	}
+	if curPath == desiredPath {
+		return nil
+	}
+	if p.walDir == "" {
+		log.Infof("moving WAL from %s to %s first and then to %s", curPath, tmpPath, desiredPath)
+	} else {
+		log.Infof("moving WAL from %s to new location %s", curPath, desiredPath)
+	}
+	// We use tmpPath here first and (if needed) mv tmpPath to desiredPath when all is copied.
+	// This allows stolon-keeper to re-read symlink dest and continue should stolon-keeper be restarted while copying.
+	log.Debugf("creating %s", tmpPath)
+	if err = os.MkdirAll(tmpPath, 0700); err != nil && !os.IsExist(err) {
+		log.Errorf("could not create new dest folder %s: %e", tmpPath, err)
+		return err
+	}
+	log.Debugf("moving WAL files from %s to %s", curPath, tmpPath)
+	if entries, err := ioutil.ReadDir(curPath); err != nil {
+		log.Errorf("could not read contents of folder %s: %e", curPath, err)
+		return err
+	} else {
+		for _, entry := range entries {
+			srcEntry := filepath.Join(curPath, entry.Name())
+			dstEntry := filepath.Join(tmpPath, entry.Name())
+			log.Debugf("moving %s to %s", srcEntry, dstEntry)
+			if err = os.Rename(srcEntry, dstEntry); err != nil {
```

Comment:
This method will only work with a single filesystem. Cross filesystem moves will throw an error. This is probably not idea since one of the main uses of this flag would be to put the WAL data on a separate volume.

Reply:
Ouch. I was testing in a docker container with one MP.
Good catch. I will submit a fix.